Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/homecard/VaccinationHomeCard.kt

```diff
@@ -0,0 +1,52 @@
+package de.rki.coronawarnapp.vaccination.ui.homecard
+
+import android.view.ViewGroup
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.VaccinationHomeCardBinding
+import de.rki.coronawarnapp.ui.main.home.HomeAdapter
+import de.rki.coronawarnapp.util.lists.diffutil.HasPayloadDiffer
+import de.rki.coronawarnapp.vaccination.core.VaccinatedPerson
+
+class VaccinationHomeCard(parent: ViewGroup) :
+    HomeAdapter.HomeItemVH<VaccinationHomeCard.Item, VaccinationHomeCardBinding>(
+        R.layout.home_card_container_layout,
+        parent
+    ) {
+
+    override val viewBinding = lazy {
+        VaccinationHomeCardBinding.inflate(layoutInflater, itemView.findViewById(R.id.card_container), true)
+    }
+
+    override val onBindData: VaccinationHomeCardBinding.(
+        item: Item,
+        payloads: List<Any>
+    ) -> Unit = { item, payloads ->
+        val curItem = payloads.filterIsInstance<Item>().singleOrNull() ?: item
+
+        personName.text = curItem.vaccinatedPerson.fullName
+        when (curItem.vaccinatedPerson.getVaccinationStatus()) {
+            VaccinatedPerson.Status.COMPLETE -> {
+                val days = curItem.vaccinatedPerson.getTimeUntilImmunity()!!.standardDays.toInt()
```

Comment:
Can we use a fallback here instead of crashing the app?

Reply:
`COMPLETE` state should always have a `duration`. this should not crash in case it complete ,and if it does we would know that there is something wrong

Reply:
Best solution here would be IMO to use a sealed class hierarchy for the Status instead of an enum and then have a `getTimeUntilImmunity()` for the `Immunity` data class. Would require some refactoring though...